src/ansys/fluent/core/services/field_data.py

```diff
@@ -638,7 +639,55 @@ class PathlinesFieldDataRequest(NamedTuple):
     zones: list | None = None
 
 
-class BaseFieldData:
+class FieldDataSource(ABC):
+    """Abstract class for field data interface."""
+
+    @abstractmethod
+    def get_surface_ids(self, surfaces: List[str | int]) -> List[int]:
+        """Get a list of surface ids based on surfaces provided as inputs."""
+        pass
+
+    @abstractmethod
+    def get_field(
+        self,
+        obj: (
+            SurfaceFieldDataRequest
+            | ScalarFieldDataRequest
+            | VectorFieldDataRequest
+            | PathlinesFieldDataRequest
+        ),
+    ) -> Dict[int | str, Dict | np.array]:
+        """Get the surface, scalar, vector or path-lines field data on a surface."""
+        pass
+
+
+class FieldTransactionSource(ABC):
+    """Abstract class for field data interface."""
+
+    @abstractmethod
+    def get_surface_ids(self, surfaces: List[str | int]) -> List[int]:
+        """Get a list of surface ids based on surfaces provided as inputs."""
+        pass
+
+    @abstractmethod
+    def add_requests(
+        self,
+        obj: (
+            SurfaceFieldDataRequest
+            | ScalarFieldDataRequest
+            | VectorFieldDataRequest
+            | PathlinesFieldDataRequest
+        ),
+        *args: SurfaceFieldDataRequest
+        | ScalarFieldDataRequest
+        | VectorFieldDataRequest
+        | PathlinesFieldDataRequest,
+    ):
+        """Add request to get surface, scalar, vector or path-lines field on surfaces."""
```

Comment:
@prmukherj request should be pluralized here. field shopuld be plural. path-lines should be singular. Use tools to make sure things are already clean.